src/components/Beneficiaries/index.tsx

```diff
@@ -75,242 +43,207 @@ const defaultBeneficiary: Beneficiary = {
+
+    const { currency, type, onChangeValue } = props;
+
+    const { formatMessage } = useIntl();
+    const dispatch = useDispatch();
+
+    /*    selectors    */
+    const beneficiaries = useSelector(selectBeneficiaries);
+    const beneficiariesAddData = useSelector(selectBeneficiariesCreate);
+    const beneficiariesAddSuccess = useSelector(selectBeneficiariesCreateSuccess);
+    const beneficiariesActivateSuccess = useSelector(selectBeneficiariesActivateSuccess);
+    const memberLevels = useSelector(selectMemberLevels);
+    const userData = useSelector(selectUserInfo);
+    const isMobileDevice = useSelector(selectMobileDeviceState);
+    /*    ---------    */
+
+    React.useEffect(() => {
+        if (currency && beneficiaries) {
+            handleSetCurrentAddressOnUpdate(beneficiaries);
         }
 
         if (!memberLevels) {
-            this.props.memberLevelsFetch();
-        }
-    }
-
-    public componentWillReceiveProps(nextProps: Props) {
-        const {
-            currency,
-            beneficiaries,
-            beneficiariesAddSuccess,
-            beneficiariesActivateSuccess,
-        } = this.props;
-
-        if ((nextProps.currency && nextProps.currency !== currency) ||
-            (nextProps.beneficiaries.length && nextProps.beneficiaries !== beneficiaries)) {
-            this.handleSetCurrentAddressOnUpdate(nextProps.beneficiaries);
+            dispatch(memberLevelsFetch());
         }
+    }, []);
+
+    const prevCurrency = usePrevious(currency);
+    const prevBeneficiaries = usePrevious(beneficiaries);
+    const prevBeneficiariesAddSuccess = usePrevious(beneficiariesAddSuccess);
+    const prevBenefiariesActivateSuccess = usePrevious(beneficiariesActivateSuccess);
 
-        if (nextProps.beneficiariesAddSuccess && !beneficiariesAddSuccess) {
-            this.handleToggleAddAddressModal();
-            this.handleToggleConfirmationModal();
+    React.useEffect(() => {
+        if ((props.currency && prevCurrency !== props.currency)
+            || (beneficiaries.length && beneficiaries !== prevBeneficiaries)) {
+            handleSetCurrentAddressOnUpdate(beneficiaries);
         }
 
-        if (nextProps.beneficiariesActivateSuccess && !beneficiariesActivateSuccess) {
-            this.handleToggleConfirmationModal();
+        if (!prevBeneficiariesAddSuccess && beneficiariesAddSuccess) {
+            setAddressModalState(false);
+            setConfirmationModalState(true);
         }
 
-        if (nextProps.beneficiariesAddSuccess && !beneficiariesAddSuccess) {
-            this.handleToggleAddAddressModal();
-            this.handleToggleConfirmationModal();
+        if (!prevBenefiariesActivateSuccess && beneficiariesActivateSuccess) {
+            setConfirmationModalState(false);
         }
+    }, [beneficiariesAddSuccess, currency, prevBeneficiariesAddSuccess, prevBenefiariesActivateSuccess, beneficiariesActivateSuccess]);
```

Comment:
1. Separate this hook to multiple `useEffect()`
2. Each hook should do one action. Example:
**This code**
```
React.useEffect(() => {
        if ((props.currency && prevCurrency !== props.currency)
            || (beneficiaries.length && beneficiaries !== prevBeneficiaries)) {
            handleSetCurrentAddressOnUpdate(beneficiaries);
        }
        ...
```
**Should be like this**
```
 React.useEffect(() => {
        if (beneficiaries) {
            handleSetCurrentAddressOnUpdate(beneficiaries);
        }
 }, [beneficiaries]);
```